train.py

```diff
@@ -235,11 +243,10 @@ def val_epoch(
     tokens = 0
     total_dist = 0
 
-    dev = torch.device(device)
-
-    for b in tqdm(dataloader):
-        batch = Batch(b[0], b[1], pad=pad_idx)
-        batch.to(dev)
+    for batch in tqdm(dataloader):
+        src = batch["source_tokens_ids"].to(device)
+        tgt = batch["target_tokens_ids"].to(device)
```

Comment:
(from here)

Reply:
PS. you also use correct "source_token_ids" in dashboard.py and predict_piece.py